src/components/menu.tsx

```diff
@@ -0,0 +1,149 @@
+import React, {MouseEvent, CSSProperties} from "react";
+import "../stylesheets/menu.scss"
+import {FontAwesomeIcon} from "@fortawesome/react-fontawesome";
+import {IconProp} from "@fortawesome/fontawesome-svg-core";
+
+/**
+ * Menu Item
+ * id: Used as keys when creating a list
+ * text: Text on the menu item 
+ * icon: Icon on the menu item
+ * onClick: Menu item onClick function handler
+ */
+export type MenuItem = {
+    id: number,
+    text: string, 
+    icon: IconProp,
+    onClick: (e: MouseEvent<HTMLElement>) => void;
+}
+
+/**
+ * Menu Props
+ * menuItems: Items in the menu
+ * originElement: HTML element that toggle the menu
+ * onClose: Menu onClose function handler
+ * position?: Location of the menu ("right" or "bottom")
+ * width?: Width of the menu (in rem)
+ * buttonHeight?: Height of each button in the menu (in rem)
+ */
+export type MenuProps = typeof Menu.defaultProps & {
+    menuItems: MenuItem[];
+    originElement: HTMLElement;
+    onClose: () => void;
+}
+
+export class Menu extends React.Component<MenuProps, {}> {
```

Comment:
Could you also add close on click outside component for Menu component? It's pretty easy to do with something like this: https://codedaily.io/tutorials/63/Create-a-Dropdown-in-React-that-Closes-When-the-Body-is-Clicked.
I've already done it for the filter selector and search bar results and it seems to work pretty well